File: src/main/java/gregtechlite/gtlitecore/mixins/Implemented.java
package gregtechlite.gtlitecore.mixins;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Implemented attributor for all mixins classes.
 * <p>
 * This annotation marked the mixins class is an adhoc resolve which implemented from GTCEu or other mods unrelease
 * contents or referenced with some external contents.
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.SOURCE)
public @interface Implemented
{

    /**
     * The source link for the implemented contents.
     *
     * @return The url format link for the implemented contents. Should use gh pull request link, issue link, e.t.c.
     */
    String[] at() default "";

}

File: src/main/kotlin/gregtechlite/gtlitecore/api/mixins/Implemented.kt
package gregtechlite.gtlitecore.api.mixins

/**
 * Marked a mixins class is an adhoc resolve which implemented from GTCEu or other mods unrelease contents.
 *
 * @param at The source link for the implemented contents.
 *
 * @see gregtechlite.gtlitecore.mixins
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.SOURCE)
annotation class Implemented(val at: Array<String>)